recipes-extended/xen/files/xen-translate.patch

```diff
@@ -615,10 +598,27 @@ PATCHES
      return xsm_ops->console_io(d, cmd);
 --- a/xen/xsm/dummy.c
 +++ b/xen/xsm/dummy.c
-@@ -158,5 +158,6 @@ void __init xsm_fixup_ops (struct xsm_op
+@@ -156,5 +156,6 @@ void __init xsm_fixup_ops (struct xsm_op
      set_to_dummy_if_null(ops, pmu_op);
      set_to_dummy_if_null(ops, dm_op);
  #endif
 +    set_to_dummy_if_null(ops, memory_translate);
      set_to_dummy_if_null(ops, xen_version);
  }
+--- a/xen/include/asm-x86/mm.h
++++ b/xen/include/asm-x86/mm.h
+@@ -229,9 +229,13 @@ struct page_info
+ #define PGT_partial       PG_mask(1, 8)
+ 
+  /* Count of uses of this frame as its current type. */
+-#define PGT_count_width   PG_shift(8)
++#define PGT_count_width   PG_shift(9)
+ #define PGT_count_mask    ((1UL<<PGT_count_width)-1)
+ 
++ /* Pinned by dom0 tools */
++#define _PGC_pinned_by_tools	PG_shift(9)
++#define PGC_pinned_by_tools	PG_mask(1, 9)
```

Comment:
Hang on...
`PGT_count_width` got incremented to accommodate for that flag, not `PGC_count_width`, and this used to set ` page->count_info` not `page->u.inuse.type_info`, so `PGC_count_width` should be bumped right?

Why was that not broken to begin with, since the original patch already did that?

Reply:
I had the same question: https://openxt.atlassian.net/browse/OXT-1258

I think it would only break when ref counts get really high, which doesn't happen in practice.

Reply:
This should probably be changed:
https://github.com/xenserver/xen-4.7.pg/blob/1b778262c1b4a3c7c8e504d50e0149fe5909d670/master/nvidia-hypercalls.patch#L487

Reply:
That is of course assuming XenServer is correct and we are wrong...